include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp

```diff
@@ -251,6 +285,25 @@ class alignment_executor_two_way
         buffer.resize(size);
         setg(std::ranges::end(buffer), std::ranges::end(buffer));
     }
+
+    //!\brief Helper function to move initialise `this` from `other`.
+    //!\copydetails seqan3::detail::alignment_executor_two_way::alignment_executor_two_way(alignment_executor_two_way && other)
+    void move_initialise(alignment_executor_two_way && other) noexcept
+    {
+        kernel = std::move(other.kernel);
+        // Get the old resource position.
+        std::ptrdiff_t old_resource_pos = std::ranges::distance(other.resource.begin(), other.resource_it);
```

Comment:
since `resource_it` is private, can you access `other.resource_it` without the visibility bug?

Reply:
It should be as it is the same type. IIRC the visibility bug was raised because the rhs operand was not the same type as the lhs operand of the equality function